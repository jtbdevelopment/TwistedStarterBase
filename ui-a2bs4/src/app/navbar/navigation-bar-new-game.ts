import {Component, Input} from '@angular/core';

@Component({
    selector: 'navigation-bar-new-game',
    template: require('./navigation-bar-new-game.html')
})
export class NavigationBarNewGameComponent {
    @Input() playerLoaded: boolean;
}
