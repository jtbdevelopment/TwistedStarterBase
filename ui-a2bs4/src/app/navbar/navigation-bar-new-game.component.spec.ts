import {inject, TestBed} from '@angular/core/testing';
import {NavigationBarNewGameComponent} from './navigation-bar-new-game.component';
import {RouterTestingModule} from '@angular/router/testing';
import {NgbModule, NgbPopoverConfig} from '@ng-bootstrap/ng-bootstrap';
import {HelpDisplayService} from '../help/help-display.service';


describe('Component:  nav bar new game component', () => {
    beforeEach(() => {
        TestBed.configureTestingModule({
            imports: [
                RouterTestingModule,
                NgbModule
            ],
            declarations: [
                NavigationBarNewGameComponent,
            ],
            providers: [
                HelpDisplayService,
                NgbPopoverConfig
            ]
        });
        TestBed.compileComponents();
    });

    it('display disabled when player not loaded', () => {
        const fixture = TestBed.createComponent(NavigationBarNewGameComponent);
        fixture.componentInstance.playerLoaded = false;
        fixture.detectChanges();
        expect(fixture).toMatchSnapshot();
    });

    it('displays toggle when player loaded', () => {
        const fixture = TestBed.createComponent(NavigationBarNewGameComponent);
        fixture.componentInstance.playerLoaded = true;
        fixture.detectChanges();
        expect(fixture).toMatchSnapshot();
    });

    it('toggling help on/off shows/hides popover', inject([HelpDisplayService], (helpService) => {
        const fixture = TestBed.createComponent(NavigationBarNewGameComponent);
        fixture.componentInstance.playerLoaded = true;
        fixture.detectChanges();
        helpService.toggleHelp();
        fixture.detectChanges();
        expect(fixture).toMatchSnapshot();
        helpService.toggleHelp();
        fixture.detectChanges();
        expect(fixture).toMatchSnapshot();
    }));

});
