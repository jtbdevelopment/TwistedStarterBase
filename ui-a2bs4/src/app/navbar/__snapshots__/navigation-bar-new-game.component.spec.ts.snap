// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component:  nav bar new game component display disabled when player not loaded 1`] = `
<app-navigation-bar-new-game
  helpDisplay={[Function HelpDisplayService]}
  helpPopovers={[Function QueryList]}
  playerLoaded="false"
>
  
  
</app-navigation-bar-new-game>
`;

exports[`Component:  nav bar new game component displays toggle when player loaded 1`] = `
<app-navigation-bar-new-game
  helpDisplay={[Function HelpDisplayService]}
  helpPopovers={[Function QueryList]}
  playerLoaded={[Function Boolean]}
>
  <ul
    class="navbar-nav"
  >
    
      
    <li
      class="nav-item"
      ng-reflect-ngb-popover="Create a brand new game."
      ng-reflect-placement="right"
      ng-reflect-popover-title="New Game"
      ng-reflect-triggers="manual"
      ngbpopover="Create a brand new game."
      placement="right"
      popovertitle="New Game"
      triggers="manual"
    >
      
          
      <a
        class="nav-link"
        href="/create"
        ng-reflect-router-link="/create"
      >
        
              
        <span
          class="fa fa-plus"
        />
        
          
      </a>
      
      
    </li>
    
  
  </ul>
  
</app-navigation-bar-new-game>
`;

exports[`Component:  nav bar new game component toggling help on/off shows/hides popover 1`] = `
<app-navigation-bar-new-game
  helpDisplay={[Function HelpDisplayService]}
  helpPopovers={[Function QueryList]}
  playerLoaded={[Function Boolean]}
>
  <ul
    class="navbar-nav"
  >
    
      
    <li
      aria-describedby="ngb-popover-1"
      class="nav-item"
      ng-reflect-ngb-popover="Create a brand new game."
      ng-reflect-placement="right"
      ng-reflect-popover-title="New Game"
      ng-reflect-triggers="manual"
      ngbpopover="Create a brand new game."
      placement="right"
      popovertitle="New Game"
      triggers="manual"
    >
      
          
      <a
        class="nav-link"
        href="/create"
        ng-reflect-router-link="/create"
      >
        
              
        <span
          class="fa fa-plus"
        />
        
          
      </a>
      
      
    </li>
    <ngb-popover-window
      _nghost-c0=""
      class="popover bs-popover-right"
      id="ngb-popover-1"
      role="tooltip"
      style="top: 0px; left: 0px;"
    >
      
      
      <div
        _ngcontent-c0=""
        class="arrow"
      />
      
      
      <h3
        _ngcontent-c0=""
        class="popover-header"
      >
        New Game
      </h3>
      <div
        _ngcontent-c0=""
        class="popover-body"
      >
        Create a brand new game.
      </div>
    </ngb-popover-window>
    
  
  </ul>
  
</app-navigation-bar-new-game>
`;

exports[`Component:  nav bar new game component toggling help on/off shows/hides popover 2`] = `
<app-navigation-bar-new-game
  helpDisplay={[Function HelpDisplayService]}
  helpPopovers={[Function QueryList]}
  playerLoaded={[Function Boolean]}
>
  <ul
    class="navbar-nav"
  >
    
      
    <li
      class="nav-item"
      ng-reflect-ngb-popover="Create a brand new game."
      ng-reflect-placement="right"
      ng-reflect-popover-title="New Game"
      ng-reflect-triggers="manual"
      ngbpopover="Create a brand new game."
      placement="right"
      popovertitle="New Game"
      triggers="manual"
    >
      
          
      <a
        class="nav-link"
        href="/create"
        ng-reflect-router-link="/create"
      >
        
              
        <span
          class="fa fa-plus"
        />
        
          
      </a>
      
      
    </li>
    
  
  </ul>
  
</app-navigation-bar-new-game>
`;
